runtime/src/main/java/org/corfudb/recovery/FastObjectLoader.java

```diff
@@ -232,20 +254,53 @@ public void updateStreamTails(long address, ILogData logData) {
      * @param entry entry to potentially apply
      * @return if we need to apply the entry.
      */
-    private boolean shouldEntryBeApplied(UUID streamId, SMREntry entry) {
+    private boolean shouldEntryBeApplied(UUID streamId, SMREntry entry, boolean isCheckpointEntry) {
+        // If white list mode, ignore everything that is not in the list or checkpoints
+        if (whiteList && !streamsToLoad.contains(streamId)) {
+            return false;
+        }
+
         // We ignore the transaction stream ID because it is a raw stream
         // We don't want to create a Map for it.
         if (streamId == runtime.getObjectsView().TRANSACTION_STREAM_ID || streamsToIgnore.contains(streamId)) {
```

Comment:
You can either add this stream to the black list, or not include it in the white list.